docker/.env.example

```diff
@@ -76,3 +76,21 @@ PUBLIC_LOGIN_PAGE_MESSAGE=
 IMMICH_WEB_URL=http://immich-web:3000
 IMMICH_SERVER_URL=http://immich-server:3001
 IMMICH_MACHINE_LEARNING_URL=http://immich-machine-learning:3003
+
+####################################################################################
+# Public Facing API Server URL - Optional
+#
+# This should point to the publicly accessible URL of the API (default is "/api")
+# Use a full URL here if you want to serve the API via a dedicated domain.
+####################################################################################
+
+IMMICH_SERVER_PUBLIC_URL=/api
```

Comment:
The environmental name is a bit confusing :-? I wonder if we have any other wording that can make it clearer to provide the example in the info box?

Reply:
Yeah, I definitely struggled to think of a good name here, largely because `IMMICH_SERVER_URL` and `IMMICH_WEB_URL` were already taken to mean the internal/private locations for those services, and this is effectly the public facing version of those.  A few alternatives I'd been considering: `IMMICH_PUBLIC_SERVER_URL`, `IMMICH_EXTERNAL_SERVER_URL`, but I am very open to suggestions!

I also tried updating the comment and example a bit, to hopefully make it clearer.  Let me know if you think that helps at all.

Reply:
I prefer `IMMICH_API_URL_EXTERNAL`, as it's for the API, not the server. Otherwise, `IMMICH_SERVER_URL_EXTERNAL` might work?

It's distinct from `IMMICH_SERVER_URL` (used for internal communication), and distinct from the web URL (which doesn't need/use a variable).
In some cases, all three URLs may be the same, but that's not necessarily always the case.

Reply:
Thanks, that sounds good to me, I've switched it to "IMMICH_API_URL_EXTERNAL".  I really like the idea of adding API in there, as that truly is what the URL represents, but I originally wasn't sure if we should introduce a new term (seems how everywhere else its still referenced as the more ambiguous "server").